libr/include/r_util/r_assert.h

```diff
@@ -0,0 +1,116 @@
+#ifndef R_ASSERT_H
+#define R_ASSERT_H
+
+#include "r_log.h"
+
+R_API void r_assert_log(RLogLevel level, const char *fmt, ...);
+
+#if defined (__GNUC__) && defined (__cplusplus)
+#define R_FUNCTION     ((const char*) (__PRETTY_FUNCTION__))
+#elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#define R_FUNCTION     ((const char*) (__func__))
+#elif defined (__GNUC__) || (defined(_MSC_VER) && (_MSC_VER > 1300))
+#define R_FUNCTION     ((const char*) (__FUNCTION__))
+#else
+#define R_FUNCTION     ((const char*) ("???"))
+#endif
+
+#define r_warn_if_reached()						\
+	do {								\
```

Comment:
there's no problem in compiling `{};` by any compiler. i have never understood the need to do this do {} while(false)} wrap.

Reply:
https://stackoverflow.com/questions/1067226/c-multi-line-macro-do-while0-vs-scope-block

Reply:
none of the explanations make any sense to me.

Reply:
It is well explained in the link. And I see no reasons to not use it, so I'll keep the do while.